contrib/storage-jdbc/src/main/java/org/apache/drill/exec/store/jdbc/clickhouse/ClickhouseDialect.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.store.jdbc.clickhouse;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.sql.SqlBasicTypeNameSpec;
+import org.apache.calcite.sql.SqlDataTypeSpec;
+import org.apache.calcite.sql.SqlDialect;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlUserDefinedTypeNameSpec;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.sql.type.BasicSqlType;
+
+/**
+ * @author feiteng.wtf
+ * @date 2021-08-08
+ */
+public class ClickhouseDialect extends SqlDialect {
```

Comment:
This is a great example of a vendor-specific implementation. Can we follow the same pattern elsewhere?

Reply:
I'm not sure if this is relevant here or not, but Calcite actually has a Clickhouse dialect.[1]. Does that "answer the mail" for this?

[1]: https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/sql/dialect/ClickHouseSqlDialect.java

Reply:
thanks, but as it is not included in the current version of calcite, so I will need to do a cherry pick on DrillCalcite1.21.0 of https://github.com/vvysotskyi/drill-calcite.git and update to 1.21.0-drill-r4, right?

Reply:
To update Calcite - yes. Please do it via PR
Here is the process how to do that: https://github.com/apache/drill/blob/master/docs/dev/Calcite.md#process-of-updating-calcite-version
You can ask me or @vvysotskyi to follow you in this

Reply:
Please have a look at https://github.com/vvysotskyi/drill-calcite/pull/4

Reply:
Your Calcite PR is merged, git tag is created and Maven artifacts are available in Jitpack:
https://jitpack.io/#vvysotskyi/drill-calcite